docs/networking/payload.md

```diff
@@ -79,7 +79,71 @@ Now that we have registered the payload we need to implement a handler. For this
 ```java
 public class ClientPayloadHandler {
     
-    public static void handleData(final MyData data, final IPayloadContext context) {
+    public static void handleDataOnMain(final MyData data, final IPayloadContext context) {
+        // Do something with the data, on the main thread
+        blah(data.age());
+    }
+}
+```
+
+Here a couple of things are of note:
+
+- The handling method here gets the payload, and a contextual object.
+- The handler of the payload method is, by default, invoked on the main thread.
+
+
+If you need to do some computation that is resource intensive, then the work should be done on the network thread, instead of blocking the main thread. This is done by setting the `HandlerThread` of the `PayloadRegistrar` to `HandlerThread#NETWORK` via `PayloadRegistrar#executesOn` before registering the payload.
+
+```java
+@SubscribeEvent
+public static void register(final RegisterPayloadHandlersEvent event) {
+    final PayloadRegistrar registrar = event.registrar("1")
+        .executesOn(HandlerThread.NETWORK); // All subsequent payloads will register on the network thread
+    registrar.playBidirectional(
+        MyData.TYPE,
+        MyData.STREAM_CODEC,
+        new DirectionalPayloadHandler<>(
+            ClientPayloadHandler::handleDataOnNetwork,
+            ServerPayloadHandler::handleDataOnNetwork
+        )
+    );
+}
+```
+
+:::note
+All payloads registered after an `executesOn` call will retain the same thread execution location until `executesOn` is called again.
+
+```java
+PayloadRegistrar registrar = event.registrar("1");
+
+registrar.playBidirectional(...); // On the main thread
+registrar.playBidirectional(...); // On the main thread
+
+// All registrar methods create a new instance, so the change
```

Comment:
```suggestion
// All registrar configuration methods create a new instance, so the change
```
"All methods" sounds like actual registration methods also create a new registrar (which is of course not the case).
Though I'm not quite happy with calling them "configuration methods" either, maybe you can think of something better.

Reply:
I couldn't think of a better descriptor, so I opted to provide more context instead by referencing the state. Let me know if this is better.